### PR TITLE
Introduce a new event to filter sites displayed on all websites dashboard

### PR DESCRIPTION
**Pull request by sgiehl**
### Description:

This PR introduces a new event `MultiSites.filterSites`, which can be used by plugins to manipulate the sites that should be displayed on the multisites dashboard.

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: plugins/MultiSites/API.php [hunk 1]
```diff
@@ -159,8 +174,14 @@ private function getSitesIdFromPattern($pattern, $_restrictSitesToLogin)
 
         // Both calls above have called Site::setSitesFromArray. We now get these sites:
         $sitesToProblablyAdd = Site::getSites();
+        $idSites = array();
```
Comment (michalkleiner) [inline plugins/MultiSites/API.php]:
```suggestion
        $idSites = [];
```

Comment (michalkleiner) [reply plugins/MultiSites/API.php]: I see now this is just code moved from another place, so fine if left as is.

File: plugins/MultiSites/API.php [hunk 2]
```diff
@@ -159,8 +174,14 @@ private function getSitesIdFromPattern($pattern, $_restrictSitesToLogin)
 
         // Both calls above have called Site::setSitesFromArray. We now get these sites:
         $sitesToProblablyAdd = Site::getSites();
+        $idSites = [];
+        if (!empty($sitesToProblablyAdd)) {
+            foreach ($sitesToProblablyAdd as $site) {
+                $idSites[] = $site['idsite'];
+            }
+        }
```
Comment (michalkleiner) [inline plugins/MultiSites/API.php]: Just a minor suggestion that doesn't need to be actioned.

We can omit the check here as `Site` always seem to handle the internal `protected static $infoSites` as an array and that is what `Site::getSites()` return, so should be safe to always iterate over it. If it's empty the foreach loop will do nothing.

```suggestion
        foreach ($sitesToProblablyAdd as $site) {
            $idSites[] = $site['idsite'];
        }
```

Comment (sgiehl) [reply plugins/MultiSites/API.php]: Had only moved the code around and didn't think too much about it 🙈

**Review by michalkleiner — APPROVED** [review]
All tests green, yey!

**Pull request merged**